test/src/basic/fileformats.cpp

```diff
@@ -1320,7 +1320,7 @@ void describeTsplibXml() {
 		auto test_simple = [](bool directed) {
 			std::string directedString = directed ? "directed" : "undirected";
 			it("reads a " + directedString + " graph with weights", [&] {
-				const string& data =
```

Comment:
I'll be a little annoying :sweat_smile: – I think you could avoid the additional string copying by just putting the data directly into the stringstream, here and below:
```
stringstream ss {ResourceFile::data("fileformats/tsplibxml/valid/" + directedString + ".xml")};
```

(As a side note, I'll also use your fix for the non-public code in the GitLab repo; that one will not be forgotten.)